File: 50dias/cpp/22.cpp
#include <iostream>
#include <list>
#include <sstream>

using namespace std;

void RemoveColchetesEspacosDaString(string& entrada) {
    erase(entrada, '[');
    erase(entrada, ']');
    erase(entrada, ' ');
}

list<int> ConverteListaStringsParaInteiros(string entrada) {
    RemoveColchetesEspacosDaString(entrada);

    list<int> inteirosEntrada;
    if (entrada.size() > 0) {
        stringstream sstr(entrada);
        while (sstr.good()) {
            string substring;
            getline(sstr, substring, ',');
            inteirosEntrada.push_back(stoi(substring));
        }
    }

    return inteirosEntrada;
}

string LeiaEntradaUsuario() {
    string entrada;

    cout << "O formato da entrada deve ser o seguinte:" << endl;
    cout << "[1, 2, 3, 4, 5, 6]" << endl;
    cout << "Digite uma lista de inteiros: ";
    getline(cin, entrada);

    return entrada;
}

void EfetuaOperacoes(list<int> inteirosEntrada) {
    int tamanhoLista = inteirosEntrada.size();
    if (tamanhoLista > 0) {
        float soma = 0;
        int menor = inteirosEntrada.front();
        int maior = menor;

        for (int i : inteirosEntrada) {
            soma += i;
            if (i < menor)
                menor = i;
            if (i > maior)
                maior = i;
        }

        float media = soma / tamanhoLista;
        cout << "O maior valor é " << maior << endl;
        cout << "O menor valor é " << menor << endl;
        cout << "A média dos valores é " << media << endl;

    }
    else
        cout << "Lista está vazia!" << endl;
}

int main()
{
    string entrada = LeiaEntradaUsuario();

    list<int> inteirosEntrada;
    inteirosEntrada = ConverteListaStringsParaInteiros(entrada);

    EfetuaOperacoes(inteirosEntrada);
    return 0;
}

File: 50dias/python/22.py

def LeiaEntradaUsuario():
    print('O formato da entrada deve ser o seguinte:')
    print('[1, 2, 3, 4, ..., 30]')
    entrada = input('Digite uma lista de inteiros: ')
    return entrada

def RemoveColcheteEspacosDaString(entrada):
    entrada = entrada.replace(']', '')
    entrada = entrada.replace('[', '')
    entrada = entrada.replace(' ', '')
    return entrada

def ConverteListaStringsParaInteiros(entrada):
    entrada = RemoveColcheteEspacosDaString(entrada)

    inteirosEntrada = []
    if (len(entrada) > 0):
        entrada = entrada.split(',')

        for palavra in entrada:
            inteirosEntrada.append(int(palavra))
    return inteirosEntrada

def EfetuaOperacoes(inteiros):
    if len(inteiros) > 0:
        soma = 0;
        menor = maior = inteiros[0]

        for i in inteiros:
            soma += i
            if i < menor:
                menor = i
            if i > maior:
                maior = i
        media = soma / len(inteiros)
        print('O maior valor é ' + str(maior))
        print('O menor valor é ' + str(menor))
        print('A média dos valores é ' + str(media))

    else:
        print('Lista está vazia!')

def main():
    entrada = LeiaEntradaUsuario();
    inteirosEntrada = ConverteListaStringsParaInteiros(entrada)
    EfetuaOperacoes(inteirosEntrada)

main()